From c0bae4137b392b74b1504ebdd9c9800acef75f9b Mon Sep 17 00:00:00 2001
From: Pawit Pornkitprasan <p.pawit@gmail.com>
Date: Wed, 19 Nov 2014 22:29:34 +0700
Subject: [PATCH 03/10] Camera2: fix video recording for older drivers

Older drivers need the preview size to be the same as the video
size. This check is done in getDesiredPreviewSize but was broken
because CameraCapabilities return an empty list instead of null.

Change-Id: If4d570923bd4809c8e651831a061414836e01a36
---
 src/com/android/camera/VideoModule.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 9a941c7..48f8b6c 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -808,7 +808,7 @@ public class VideoModule extends CameraModule
      */
     private static Point getDesiredPreviewSize(CameraCapabilities capabilities,
           CamcorderProfile profile, Point previewScreenSize) {
-        if (capabilities.getSupportedVideoSizes() == null) {
+        if (capabilities.getSupportedVideoSizes().size() == 0) {
             // Driver doesn't support separate outputs for preview and video.
             return new Point(profile.videoFrameWidth, profile.videoFrameHeight);
         }
-- 
2.7.4

